ts/Stock/Indicators/CCI/CCIIndicator.ts

```diff
@@ -0,0 +1,218 @@
+/* *
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ * */
+
+'use strict';
+
+/* *
+ *
+ * Imports
+ *gul
+ * */
+
+import type IndicatorValuesObject from '../IndicatorValuesObject';
+import type LineSeries from '../../../Series/Line/LineSeries';
+import type {
+    CCIOptions,
+    CCIParamsOptions
+} from './CCIOptions';
+import type CCIPoint from './CCIPoint';
+import BaseSeries from '../../../Core/Series/Series.js';
+const {
+    seriesTypes: {
+        sma: SMAIndicator
+    }
+} = BaseSeries;
+import U from '../../../Core/Utilities.js';
+const {
+    isArray,
+    extend,
+    merge
+} = U;
+
+/* eslint-disable valid-jsdoc */
+// Utils:
+/**
+ * @private
+ */
+function sumArray(array: Array<number>): number {
+    return array.reduce(function (prev, cur): number {
+        return prev + cur;
+    }, 0);
+}
+
+/**
+ * @private
+ */
+function meanDeviation(arr: Array<number>, sma: number): number {
+    var len = arr.length,
+        sum = 0,
+        i: number;
+
+    for (i = 0; i < len; i++) {
+        sum += Math.abs(sma - (arr[i]));
+    }
+
+    return sum;
+}
+
+/* eslint-enable valid-jsdoc */
+
+/* *
+ *
+ * Class
+ *
+ * */
+
+/**
+ * The CCI series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.cci
+ *
+ * @augments Highcharts.Series
+ */
+class CCIIndicator extends SMAIndicator {
+    /**
+     * Commodity Channel Index (CCI). This series requires `linkedTo` option to
+     * be set.
+     *
+     * @sample stock/indicators/cci
+     *         CCI indicator
+     *
+     * @extends      plotOptions.sma
+     * @since        6.0.0
+     * @product      highstock
+     * @requires     stock/indicators/indicators
+     * @requires     stock/indicators/cci
+     * @optionparent plotOptions.cci
+     */
+    public static defaultOptions: CCIOptions = merge(SMAIndicator.defaultOptions, {
+        params: {
+            period: 14
+        }
+    } as CCIOptions);
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public data: Array<CCIPoint> = void 0 as any;
+    public points: Array<CCIPoint> = void 0 as any;
+    public options: CCIOptions = void 0 as any;
+
+    public getValues<TLinkedSeries extends LineSeries>(
+        series: TLinkedSeries,
+        params: CCIParamsOptions
+    ): (IndicatorValuesObject<TLinkedSeries>|undefined) {
```

Comment:
`getValues` is a function. You have two options: Add a generic "Functions" comment or remove the "Properties" one. This is important to distinguish, because property assignments in class pattern go on the instance, functions on the prototype.